### PR TITLE
add code numbering

### PR DESCRIPTION
**Pull request by PoorTable**
### Description

[Issue](https://nozbe.app/teams/Xi3SI50NCpPkVxNz/task_id/qW1w1ZCIJjHXCmjB) - After upgrade ability to add code lines was lost

### Changes

Line numbers added back

### Considerations

There is one issue with MDX, we can't pass props for some reason, nothing didn't work, I tried to change the prop to `key=value` (this approach described [here](https://github.com/syntax-tree/mdast#code)) but it din't help.

### Demo

The 'demo' changed articles to new implementation:
https://deploy-preview-26--brains-and-beards-com.netlify.app/blog/2022-dont-rerender-all-flatlist-items/
https://deploy-preview-26--brains-and-beards-com.netlify.app/blog/failing-gracefully/
https://deploy-preview-26--brains-and-beards-com.netlify.app/blog/i18n-in-react-native-apps/
https://deploy-preview-26--brains-and-beards-com.netlify.app/blog/flutter-vs-react-native-web/
https://deploy-preview-26--brains-and-beards-com.netlify.app/blog/how-to-add-redux-persist-to-flutter-app/

Screenshot 2023-05-22 at 07 56 43

### CODE REVIEW

File: src/components/Blog/PrismSyntaxHighlight.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ const calculateLinesToHighlight = meta => {
   }
```
Comment (sharnik) [inline src/components/Blog/PrismSyntaxHighlight.tsx:13]: We have [code for the option to highlight specific lines](https://github.com/brains-and-beards/open-brainsandbeards-com/blob/main/src/components/Blog/PrismSyntaxHighlight.tsx#L7-L11) implemented in a similar way as here: https://coffeeaddict.dev/gatsby-mdx-prism/ using `python{1,2,20-22}` syntax. Can you double-check if it (still?) works? I don't think we're using it now, but if we have it, it'd be good to keep it.

File: src/templates/blogTemplate.tsx [hunk 2]
```diff
@@ -64,10 +64,17 @@ export const query = graphql`
...
-    return className ? (
-      <PrismSyntaxHighlight className={className}>{children}</PrismSyntaxHighlight>
-    ) : (
-      <code>{children}</code>
+    if (!className) {
+      return <code>{children}</code>
+    }
+
+    const [language, countLineNumbers] = className.split(':')
+    const shouldCountLineNumbers = countLineNumbers?.split('=')[1] === 'true'
```
Comment (sharnik) [inline src/templates/blogTemplate.tsx]: Do I see it correctly that you never check for the key here (assume line numbering is the only option we could pass)? And `tsx:blahX=true` would number the lines just the same as `tsx:numberLines=true`?

This should rather be parsed into a hash where we can check any option's value and not depend on option order.

File: src/components/Blog/PrismSyntaxHighlight.tsx (hunk 1 above)
Comment (PoorTable) [reply src/components/Blog/PrismSyntaxHighlight.tsx:13]: We keep it, but I'm not sure if it works as expected (style is added to the line but I'm not sure if something changed)

File: src/templates/blogTemplate.tsx (hunk 2 above)
Comment (PoorTable) [reply src/templates/blogTemplate.tsx]: Changed, now we creating object

File: src/templates/blogTemplate.tsx [hunk 3]
```diff
@@ -62,12 +62,31 @@ export const query = graphql`
   }
 `
 
+const createRecordtFromString = string => {
```
Comment (sharnik) [inline src/templates/blogTemplate.tsx]: Typo: Recordt -> Record

Comment (PoorTable) [reply src/templates/blogTemplate.tsx]: fixed

**Review by sharnik — APPROVED** [review]
Looks good. Let's adjust the existing blog posts and merge it 👍

**Pull request merged**